metadata/registry_test.go

```diff
@@ -0,0 +1,124 @@
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "http://www.github.com/ksonnet/parts",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "https://www.github.com/ksonnet/parts",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "github.com/ksonnet/parts",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "www.github.com/ksonnet/parts",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+
+		&ghRegistryGetSuccess{
+			source: "http://github.com/ksonnet/parts/tree/master/incubator",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "http://github.com/ksonnet/parts/tree/master/incubator/",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "http://www.github.com/ksonnet/parts/tree/master/incubator",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "https://github.com/ksonnet/parts/tree/master/incubator",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "github.com/ksonnet/parts/tree/master/incubator",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "www.github.com/ksonnet/parts/tree/master/incubator",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/incubator/registry.yaml",
+		},
+
+		&ghRegistryGetSuccess{
+			source: "http://github.com/ksonnet/parts/blob/master/registry.yaml",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "http://www.github.com/ksonnet/parts/blob/master/registry.yaml",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "https://github.com/ksonnet/parts/blob/master/registry.yaml",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "github.com/ksonnet/parts/blob/master/registry.yaml",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+		&ghRegistryGetSuccess{
+			source: "www.github.com/ksonnet/parts/blob/master/registry.yaml",
+			target: "https://raw.githubusercontent.com/ksonnet/parts/master/registry.yaml",
+		},
+	}
+
+	for _, success := range successes {
+		success.Test(t)
+	}
+}
+
+//
+// TODO: Add failure tests, like:
```

Comment:
Might want to delete the below commented out tests before submitting, or add the tests.